components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/step/impl/DefaultStepHandler.java

```diff
@@ -1058,4 +1070,31 @@ private AuthenticatorConfig getAuthenticatorConfig() {
         }
         return authConfig;
     }
+
+    /**
+     * Check whether the user should be redirected to the retry.jsp page when the user's account is locked.
+     * This decision is taken based on three configuration options, redirectToMultiOptionPageOnFailure,
+     * showAuthFailureReasonOnLoginPage and redirectToRetryPageOnAccountLock.
+     *
+     * @param context  Authentication context.
+     * @return boolean Whether the user should be directed to retry.jsp page or not.
+     */
+    private boolean isRedirectionToRetryPage(AuthenticationContext context) {
+
+        boolean sendToMultiOptionPage = context.isSendToMultiOptionPage();
+        if (!sendToMultiOptionPage) {
+            Map<String, String> parameterMap = getAuthenticatorConfig().getParameterMap();
+            if (MapUtils.isNotEmpty(parameterMap)) {
+                String showAuthFailureReasonOnLoginPage =
+                        parameterMap.get(FrameworkConstants.SHOW_AUTH_FAILURE_REASON_ON_LOGIN_PAGE_CONF);
+                if (Boolean.parseBoolean(showAuthFailureReasonOnLoginPage)) {
+                    return false;
+                }
+                String redirectToRetryPageOnAccountLock =
+                        parameterMap.get(FrameworkConstants.REDIRECT_TO_RETRY_PAGE_ON_ACCOUNT_LOCK_CONF);
```

Comment:
What is the behavior if we are not configure REDIRECT_TO_RETRY_PAGE_ON_ACCOUNT_LOCK_CONF, SHOW_AUTH_FAILURE_REASON_ON_LOGIN_PAGE_CONF values in config files, could this throw an error? I think we should check the existence of the key first and then only get this.

Reply:
When the key REDIRECT_TO_RETRY_PAGE_ON_ACCOUNT_LOCK_CONF is not available   
String redirectToRetryPageOnAccountLock = parameterMap.get(FrameworkConstants.REDIRECT_TO_RETRY_PAGE_ON_ACCOUNT_LOCK_CONF) is evaluated to null and (Boolean.parseBoolean(redirectToRetryPageOnAccountLock) is evaluated to false.